streams/src/test/java/org/apache/kafka/streams/integration/utils/EmbeddedSingleNodeKafkaCluster.java

```diff
@@ -0,0 +1,135 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streams.integration.utils;
+
+import org.apache.curator.test.InstanceSpec;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Properties;
+
+
+/**
+ * Runs an in-memory, "embedded" Kafka cluster with 1 ZooKeeper instance and 1 Kafka broker.
+ */
+public class EmbeddedSingleNodeKafkaCluster {
+
+    private static final Logger log = LoggerFactory.getLogger(EmbeddedSingleNodeKafkaCluster.class);
+
+    private final ZooKeeperEmbedded zookeeper;
+    private final KafkaEmbedded broker;
+
+    /**
+     * Creates and starts a Kafka cluster.
+     */
+    public EmbeddedSingleNodeKafkaCluster() throws Exception {
+        this(new Properties());
+    }
+
+    /**
+     * Creates and starts a Kafka cluster.
+     *
+     * @param brokerConfig Additional broker configuration settings.
+     */
+    public EmbeddedSingleNodeKafkaCluster(Properties brokerConfig) throws Exception {
+        log.debug("Initiating embedded Kafka cluster startup");
+        int zkPort = InstanceSpec.getRandomPort();
+        log.debug("Starting a ZooKeeper instance on port {} ...", zkPort);
+        zookeeper = new ZooKeeperEmbedded(zkPort);
+        log.debug("ZooKeeper instance is running at {}", zookeeper.connectString());
+
+        Properties effectiveBrokerConfig = effectiveBrokerConfigFrom(brokerConfig, zookeeper);
+        log.debug("Starting a Kafka instance on port {} ...", effectiveBrokerConfig.getProperty("port"));
+        broker = new KafkaEmbedded(effectiveBrokerConfig);
+        broker.start();
+        log.debug("Kafka instance is running at {}, connected to ZooKeeper at {}",
+            broker.brokerList(), broker.zookeeperConnect());
+    }
+
+    private Properties effectiveBrokerConfigFrom(Properties brokerConfig, ZooKeeperEmbedded zookeeper) {
+        Properties effectiveConfig = new Properties();
+        effectiveConfig.put("zookeeper.connect", zookeeper.connectString());
+        int brokerPort = InstanceSpec.getRandomPort();
```

Comment:
Any reason why we don't just use 0 as the port and let a random port be assigned?